doc/sphinx/addendum/type-classes.rst

```diff
@@ -439,6 +439,26 @@ Summary of the commands
    + Use the :cmd:`Typeclasses eauto` command to customize the behavior of
      this tactic.
 
+   .. _TypeclassesEautoBestEffort:
+
+   :n:`best_effort`
+     If the `best_effort` option is given, and resolution fails, `typeclasses eauto`
+     will return the first partial solution where all remaining subgoals fall into one
+     of these two categories:
+
+     - Stuck goals: the head of the goal has at least one associated declared mode
+       and the constraint does not match any mode declared for its head. These goals
+       are put on the shelf.
+
+     - Mode failures: the head of the constraint has at least one matching declared mode,
+       but the constraint couldn't be solved. These goals are left as subgoals of
+       :n:`typeclasses eauto best_effort`.
+
+     This option is always used when calling typeclass resolution during type inference:
+     it allows to construct a partial solution to the goals and give more informative
+     error messages. It can be used the same way in interactive proofs: it allows to see which
+     instances are missing for a typeclass resolution to succeed.
```

Comment:
```suggestion
     This option is always used for typeclass resolution during type inference:
     it allows constructing a partial solution for the goals and giving more informative
     error messages. It can be used the same way in interactive proofs: it lets users see which
     instances are preventing typeclass resolution from succeeding.
```

"This option is always used" - "optional" and "always" appear a bit inconsistent.  Is there better wording?

I think I'd move this up and combine with the paragraph at line 445 something like this:

     If the `best_effort` option is given and resolution fails, `typeclasses eauto`
     returns the first partial solution it finds and generates an informative error
     message.  This option ...

     The unsolved subgoals fall into one of these categories:   ...